src/test/java/com/iexec/core/replicate/ReplicateServiceTests.java

```diff
@@ -522,39 +524,32 @@ void shouldNotUpdateReplicateStatusSinceNoMatchingReplicate() {
                 .build();
 
         replicatesService.updateReplicateStatus(CHAIN_TASK_ID, WALLET_WORKER_2, statusUpdate);
-        Mockito.verify(replicatesRepository, Mockito.times(0))
-                .save(any());
-        Mockito.verify(applicationEventPublisher, Mockito.times(0))
-                .publishEvent(any());
+        verifyNoInteractions(applicationEventPublisher);
     }
 
     @Test
     void shouldNotUpdateReplicateStatusSinceInvalidWorkflowTransition() {
         Replicate replicate = new Replicate(WALLET_WORKER_1, CHAIN_TASK_ID);
         replicate.updateStatus(CONTRIBUTED, ReplicateStatusModifier.WORKER);
         ReplicatesList replicatesList = new ReplicatesList(CHAIN_TASK_ID, Collections.singletonList(replicate));
-
-        when(replicatesRepository.findByChainTaskId(CHAIN_TASK_ID)).thenReturn(Optional.of(replicatesList));
+        replicatesRepository.save(replicatesList);
 
         ReplicateStatusUpdate statusUpdate = ReplicateStatusUpdate.builder()
                 .modifier(WORKER)
                 .status(REVEALED)
                 .build();
 
         replicatesService.updateReplicateStatus(CHAIN_TASK_ID, WALLET_WORKER_1, statusUpdate);
-        Mockito.verify(replicatesRepository, Mockito.times(0))
-                .save(any());
-        Mockito.verify(applicationEventPublisher, Mockito.times(0))
-                .publishEvent(any());
+        verifyNoInteractions(applicationEventPublisher);
```

Comment:
the control on the repo has been removed. no need to check anything anymore? via a count?